deployment/cdk/opensearch-service-migration/lib/lambda/msk-public-endpoint-handler.ts

```diff
@@ -0,0 +1,172 @@
+            }
+        );
+    } catch (e) {
+        console.error(e);
+    }
+}
+
+async function describeClusterV2(): Promise<any> {
+    const mskClient = new KafkaClient({})
+    const params = {
+        ClusterArn: process.env.MSK_ARN
+    }
+    const command = new DescribeClusterV2Command(params)
+    let result
+    try {
+        console.log("Starting describe cluster call")
+        await mskClient.send(command).then(
+            (data) => {
+                console.log("Describe cluster response: " + JSON.stringify(data))
+                result = data
+            },
+            (error) => {
+                console.log("Describe cluster error: " + JSON.stringify(error))
+            }
+        );
+    } catch (e) {
+        console.error(e);
+    }
+    return result
+}
+
+async function getBootstrapBrokers(): Promise<any> {
+    const mskClient = new KafkaClient({})
+    const params = {
+        ClusterArn: process.env.MSK_ARN
+    }
+    const command = new GetBootstrapBrokersCommand(params)
+    let result
+    try {
+        console.log("Starting get brokers call")
+        await mskClient.send(command).then(
+            (data) => {
+                console.log("Get brokers response: " + JSON.stringify(data))
+                result = data
+            },
+            (error) => {
+                console.log("Get brokers error: " + JSON.stringify(error))
+            }
+        );
+    } catch (e) {
+        console.error(e);
+    }
+    return result
+}
+
+async function invokeNextLambda(payload: any, functionName: string) {
+    const lambdaClient = new LambdaClient({})
+    const params = {
+        FunctionName: functionName,
+        Payload: payload,
+        // Perform async request which does not wait on response
+        InvocationType: "Event"
+    }
+    const command = new InvokeCommand(params)
+    try {
+        console.log("Starting invoke lambda call")
+        await lambdaClient.send(command).then(
+            (data) => {
+                console.log("Invoke lambda response: " + JSON.stringify(data))
+                data.$metadata.httpStatusCode
+
+            },
+            (error) => {
+                console.log("Invoke lambda error: " + JSON.stringify(error))
+            }
+        );
+    } catch (e) {
+        console.error(e);
+    }
+}
+
+export const handler = async (event: any, context: Context): Promise<void> => {
+    console.log('Lambda is invoked with event:' + JSON.stringify(event));
+    console.log('Lambda is invoked with context:' + JSON.stringify(context));
+
+    let maxAttempts = process.env.MAX_ATTEMPTS ? parseInt(process.env.MAX_ATTEMPTS) : undefined
+    if (!maxAttempts) {
+        console.log('Max attempts not provided by environment or unable to be parsed, defaulting to 3 attempts')
+        maxAttempts = 3
+    }
+    if (!process.env.MSK_ARN) {
+        throw Error(`Missing at least one required environment variable [MSK_ARN: ${process.env.MSK_ARN}]`)
```

Comment:
Currently it would fail the lambda and you may or may not see in terminal. I've adjusted to use callback so that the pipeline is cutoff if encountered